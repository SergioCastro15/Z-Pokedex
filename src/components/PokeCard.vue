<template>
  <div>
    <b-row>
      <b-card
        v-for="(sprite, index) in spritesPokemon"
        :key="index"
        title="Type of pokemon"
        :img-src="sprite"
        style="width: 17rem;"
        class="mx-1 my-2"
      >
        <b-card-text>
          {{ isShiny(index) }}
        </b-card-text>
      </b-card>
    </b-row>
  </div>
</template>

<script>
export default {
  name: 'PokeCard',
  props: {
    pokemon: {
      type: Object,
      required: true,
    },
  },
  computed: {
    spritesPokemon() {
      const sprites = Object.keys(this.pokemon || {});

      return sprites.filter((sprite) => this.pokemon[sprite] !== null)
        .map((key) => this.pokemon[key]);
    },
  },
  methods: {
    isShiny(index) {
      return index % 2 === 0 ? 'Normal' : 'Shiny';
    },
  },
};
</script>
