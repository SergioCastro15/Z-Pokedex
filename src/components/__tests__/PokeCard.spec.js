import { shallowMount } from '@vue/test-utils';
import PokeCard from '../PokeCard.vue';

describe('PokeCard component', () => {
  it('render', () => {
    const wrapper = shallowMount(PokeCard);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should return an array with the elements diferent to null', () => {
    const localThis = {
      pokemon: {
        front_pokemon: 'foo',
        back_pokemon: null,
        front_pokemon_shiny: 'foo2',
        back_pokemon_shiny: 'foo3',
      },
    };
    const response = PokeCard.computed.spritesPokemon.call(localThis);
    expect(response.length).toBe(3);
  });

  it('should return the type of pokemon', () => {
    const responseShiny = PokeCard.methods.isShiny(3);
    expect(responseShiny).toBe('Shiny');

    const response = PokeCard.methods.isShiny(4);
    expect(response).toBe('Normal');
  });
});
