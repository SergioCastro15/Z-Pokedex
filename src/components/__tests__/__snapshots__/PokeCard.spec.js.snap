// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeCard component render 1`] = `
<div>
  <b-row />
</div>
`;
